### DISCUSSION TITLE
EnvKey Source Installation Docker Containers

### DISCUSSION THREAD

**Discussion by green2jello** [envkey/envkey#91] [Q&A]
I was having trouble getting the [IS_DOCKER var to set in the installation script](https://github.com/envkey/envkey-source/blob/cfab5c6892fbf97ae8d93df2abb4445446816be2/install.sh#L42) using an ubuntu:22.04 container during build time.

It looks like with current ubuntu docker images that the result from listing out the cgroup does not include anything with docker:
```
$ docker run --rm -it ubuntu:22.04 /bin/bash -c "cat /proc/1/cgroup"
0::/
```

If there is a desire to still use something in the `proc` vicinity to confirm that you are in a docker container still I think that (`/.dockerenv` is a non permanent hack from the buildkit devs from what I remember) is using `/proc/self/mountinfo`. My suggested changes for that if statement is:

`if [ $(cat /proc/1/cgroup 2> /dev/null | grep docker | wc -l) -gt 0 ] || [ -f /.dockerenv ] || [ $(cat /proc/self/mountinfo 2> /dev/null | grep docker | wc -l) -gt 0 ]; then`

I confirmed this installation script worked still with ubuntu:22.04, python:3, rocklylinux:9, and alpine:latest. Otherwise the base installation script would route towards the else statement with the sudo call.

**Comment by danenania** [comment]
Thanks @green2jello, that makes sense and seems like a better approach. Could you put up those changes as a PR?
